frontend/src/metabase/public/lib/code-templates.ts

```diff
@@ -1,16 +1,38 @@
 import type {
   CodeSampleParameters,
   EmbeddingDisplayOptions,
-  EmbeddingParameters,
+  EmbeddingParametersValues,
 } from "./types";
 import { optionsToHashParams } from "./embed";
 
+function getIframeQuerySource(displayOptions: EmbeddingDisplayOptions) {
+  return JSON.stringify(
+    optionsToHashParams(
+      removeDefaultValueParameters(displayOptions, {
+        theme: "light",
+        hide_download_button: false,
+      }),
```

Comment:
I think this is better. So we keep the value as we want to store in React state e.g. `hide_download_button: false` but when using it we convert it to the value that's appropriate to each representation in this case we don't want to show `theme` and `hide_down_button` in URL parameters when their values are default values.